scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -46,55 +60,130 @@ namespace <%= namespace %> {
         /// \endcode
         public Cart(ShopifyClient client) {
             Client = client;
-            _LineItems = new CartLineItems();
+            _LineItems = new CartLineItems(OnDeleteLineItem);
         }
 
-        /// <summary>
-        /// Returns a link which can be used to check out this cart's line items via a web browser.
-        /// </summary>
-        /// <param name="note">can be used to create a special note for the merchant for this order</param>
-        /// \code
-        /// // Example that creates a checkout link and opens the url in the Browser
-        /// string linkToCheckout = ShopifyBuy.Client().Cart().GetWebCheckoutLink();
-        ///
-        /// Application.OpenURL(linkToCheckout);
-        /// \endcode
-        public string GetWebCheckoutLink(string note = null) {
-            StringBuilder url = new StringBuilder();
-            bool hasLineItem = false;
+        public delegate void GetWebCheckoutLinkSuccess(string url);
+        public delegate void GetWebCheckoutLinkFailure(List<string> graphQLErrors, string httpError);
+
+        public void GetWebCheckoutLink(GetWebCheckoutLinkSuccess success, GetWebCheckoutLinkFailure failure) {
+            CheckoutSave((errors, httpError) => {
+                if (errors != null || httpError != null) {
+                    failure(errors, httpError);
+                } else {
+                    success(CurrentCheckout.webUrl());
+                }
+            });
+        }
+
+        private void CheckoutSave(Action<List<string>, string> callback) {
```

Comment:
hmm i prefer delegates because they are more explicit, but im ok with action if thats what you wanna do for private callbacks. Its more verbose, but i find it nicer to write when using resharper or intellisense or whatever because it gives you the context of the information, including the named variables so that its clear whats coming back in that callback and how it should be used...

BUT

Implementation is implementation, we can always go a different direction so Action is fine here... Want to agree that we should use delegate types for interfaces and we use Action/doesnt matter for private stuff? WDYT?